### PR TITLE
Version bumps

### PR DESCRIPTION
**Pull request by brandonvin**
A bit of spring cleaning in the form of version bumps:
* Bump cljstyle version to 0.15.0.
* Change the 2 supported Spark versions in tests to 3.1.x and 3.2.x.
* Drop the empty sparkplug-ml project.
* Fix mismatched snapshot versions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,13 +15,13 @@ executors:
...
+    executor: clojure-java-11
     steps:
       - checkout
       - run:
           name: Install cljstyle
           environment:
-            CLJSTYLE_VERSION: 0.14.0
+            CLJSTYLE_VERSION: 0.15.0
           command: |
             wget https://github.com/greglook/cljstyle/releases/download/${CLJSTYLE_VERSION}/cljstyle_${CLJSTYLE_VERSION}_linux.tar.gz
```
Comment (greglook) [inline .circleci/config.yml:26]: You should also switch these commands to use the `.zip` package, the tarball is deprecated and won't be available in future releases.

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,42 +32,80 @@ jobs:
...
           keys:
-            - v1-test-{{ checksum "project.clj" }}
-            - v1-test-
+            - v1-test-spark-3.1-java-8-{{ checksum "project.clj" }}
+            - v1-test-spark-3.1-java-8-
       - run:
           name: Test projects
-          command: lein monolith each do clean, check, install, test
+          command: |
+            lein -version
```
Comment (greglook) [inline .circleci/config.yml:46]: Prefer the task form of this command:
```suggestion
            lein version
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -32,42 +32,80 @@ jobs:
...
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-test-spark-3.2-java-8-{{ checksum "project.clj" }}
+            - v1-test-spark-3.2-java-8-
+      - run:
+          name: Test projects
+          command: |
+            lein -version
```
Comment (greglook) [inline .circleci/config.yml:64]: Same
```suggestion
            lein version
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ version: 2.1
 
 # Common executor configuration
 executors:
-  clojure:
```
Comment (drewinglis) [inline .circleci/config.yml:5]: You might be able to simplify this config using [matrix jobs](https://circleci.com/blog/circleci-matrix-jobs/).

**Review by drewinglis — APPROVED** [review]

File: .circleci/config.yml (hunk 4 above)
Comment (brandonvin) [reply .circleci/config.yml:5]: Thanks for the heads up, I'll play around with that later.

File: .circleci/config.yml [hunk 5]
```diff
@@ -15,13 +15,13 @@ executors:
 # Job definitions
 jobs:
   style:
-    executor: clojure
+    executor: clojure-java-11
```
Comment (greglook) [inline .circleci/config.yml:18]: If the combinatorial versions start getting out of hand, we should look into https://circleci.com/docs/2.0/dynamic-config/

**Pull request merged**